### ISSUE TITLE
(Feature Request) add tags to bookmarks.

### ISSUE THREAD

**Issue by 5c0tt-b0t** [xwmx/nb#6]
First... Thank you! Its great!!
I think a great feature would be tags for bookmarks, this will keep them all organised.
Thank you again!
**EDIT:** noticed notes already have notebooks! 👍 
Could we add bookmarks to these too?
E.g: have a "bash" notebook containing both bash-related notes & bookmarks?
**EDIT EDIT!** Im dumb, I found you already can do  this by using: `nb use <notebook>` to switch the current notebook, can I save to another notebook not in currently in use though? Without switching to it first? something like: `nb (URL) (AnotherNoteBook)` ?

**Comment by xwmx** [comment]
Glad you find it useful!

Regarding tags, bookmarks can be tagged using the `--tags` option. Tags are converted into hashtags:

```bash
nb https://example.com --tags tag1,tag2
```

This creates a bookmark with a `## Tags` section:

```markdown
# Example Title (example.com)

<https://example.com>

## Description

Example description.

## Tags

#tag1 #tag2

...[omitted]...
```

Search for tagged bookmarks (and notes containing hashtags) with `nb search` (with quotes around the hashtag):

```bash
nb search "#tag1"
```

`nb` notebooks are also a really good way to organize bookmarks, and, yes, you can add a bookmark to a different notebook than the one currently in use. To do this, pass the notebook name followed by a colon, then a space, then the URL and any options:

```bash
# create a new bookmark in a notebook named "sample"
nb sample: https://example.com --tags tag1,tag2
```

There's more information about running commands in different notebooks in the [Notebooks section of the README](https://github.com/xwmx/nb#-notebooks).

Let me know if you have additional questions!

**Comment by xwmx** [comment]
Prompted by your questions, I just made some improvements to the behavior around notebooks, available in the latest version, 4.6.1.

Among other things, when a new notebook is created and the contents of the empty notebook are listed, the messages now display the commands with the notebook prefix and colon, which should make it easier to see how to quickly add bookmarks to notebooks. For example:

```bash
> nb notebooks create "example"
Added notebook: example

> nb example:
0 notes.

Add a note:
  nb example:add
Add a bookmark:
  nb example: <url>
Import a file:
  nb example:import (<path> | <url>)
Help information:
  nb help
```

Also, regarding:

> E.g: have a "bash" notebook containing both bash-related notes & bookmarks?

Yes, definitely, this is exactly how `nb` is designed to be used. Then you can use the colon prefix at any time to save and view notes and bookmarks in that notebook. For example:

```bash
# save a bookmark to the "bash" notebook
> nb bash: http://example.com

# open bookmark id 42 in the "bash" notebook in you browser
> nb bash:open 42

# save a code snippet to the "bash" notebook from the clipboard on linux
> xclip -o | nb bash:add example.sh

# save a code snippet to the "bash" notebook from the clipboard on macOS
> pbpaste | nb bash:add example.sh

# add a note to the "bash" notebook
> nb bash:add "Example note content."

# view a note from the "bash" notebook
> nb show bash:56

# edit a note from the "bash" notebook
> nb edit bash:56
```

`nb` works well with a bunch of notebooks, so they are great for organizing by topic. Tags have the feature of being searchable across notebooks, so they end up being complementary organizational systems.

I hope this helps! I'll mark this as closed, but feel free to re-open!

**Comment by 5c0tt-b0t** [comment]
WOW :O
I completely missed your reply and only noticed this today in the CLi that you changed things!!!
**THANK YOU AGAIN!** for such an yet another amazing bash utility. Its appreciated!!!
